pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -144,6 +143,14 @@ def results(self) -> Optional[Union[_EVALUATE_OUTPUT, _PREDICT_OUTPUT]]:
         """
         return self._results
 
+    @property
+    def plugin_restores_model(self) -> bool:
+        return False
+
+    @property
+    def plugin_restores_optimizers(self) -> bool:
+        return False
+
```

Comment:
instead of having split responsibilities between the checkpoint connector and plugin, I'd prefer the accelerator/plugin always owning the responsibilities for saving/loading both model and optimizer states.

- for oncoming items like https://github.com/PyTorchLightning/pytorch-lightning/issues/6691
- for easier debugging/codebase comprehensibility (rather than hopping through plugin settings)

wdyt?

Reply:
yes, I was experimenting and it doesn't work well with these properties. Looking for a different solution now. 
But I will generally try to push back on bloating the training type plugin with such responsibilities. Currently, there is the deepspeed plugin that is calling the hooks on the LM. If we aim towards a consistent hook call order, we can't have plugins override it and implement arbitrary logic. Also it leads to duplication and I am worried that people will see this pattern and repeat it. 

I'm looking at two possible now hooks in the plugin now:

```python
    def load_model_state_dict(self, checkpoint: Mapping[str, Any]) -> None:
        self.lightning_module.load_state_dict(checkpoint["state_dict"])

    def load_checkpoint_file(self, checkpoint_path: Union[str, Path]) -> Dict[str, Any]:
        return pl_load(checkpoint_path, map_location=(lambda storage, loc: storage))
```
the checkpoint connector would still lhave full control in which order these are called.

Reply:
@ananthsub does this look better to you: https://github.com/PyTorchLightning/pytorch-lightning/pull/7928
(split out changes from this PR)